cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -340,8 +340,8 @@ public void received(StreamWriterHeartbeat heartbeat, NotificationContext notifi
         }, heartbeatsSubscriptionExecutor);
       } catch (NotificationFeedException e) {
         // Most probably, the dataset service is not up. We retry
-        LOG.warn("Could not subscribe to heartbeats feed", e);
-        TimeUnit.MILLISECONDS.sleep(200);
+        LOG.warn("Could not subscribe to heartbeats feed. {}", e.getMessage());
```

Comment:
I just didn't want to log the full exception, as, like you said, the logs could be flooded with those.